restartmanager/restartmanager.go

```diff
@@ -20,12 +20,7 @@ const (
 var ErrRestartCanceled = errors.New("restart canceled")
 
 // RestartManager defines object that controls container restarting rules.
-type RestartManager interface {
-	Cancel() error
-	ShouldRestart(exitCode uint32, hasBeenManuallyStopped bool, executionDuration time.Duration) (bool, chan error, error)
-}
-
-type restartManager struct {
```

Comment:
I'm guessing it was using an interface (exported) and `restartManager` type (not exported) to enforce that it had to be initialised through `.New()`. Not sure if that's worth the extra complexity though (but open to feedback)

Reply:
Hmmm do any of the linters we're using have a way to help enforce that?  I guess maybe we're probably not even using this in enough places to really worry about it?

Reply:
Yeah, I don't think we should worry about this. It's (somewhat) inherit to how Go work, and it not being "really" OO; I guess we have any types that are only functional / usable when using a constructor (or being sure all fields are set), and nothing really prevents someone from just `foo := SomeType{}` without setting those fields.

I think the change in this PR outweighs that theoretical "issue".